### ISSUE TITLE
feat[dev]: ✨ implement A11y improvements to sd-notification

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#1482]
## Description
this task describes the Storybook implementation for the sd-notification component as part of the A11y improvements outlined in Epic https://github.com/solid-design-system/solid/issues/1465.

A11y improvements for sd-notification can be found here: [sd-notification Notes](https://www.notion.so/matuzo/sd-notification-10ef3f6e1bc08069a42fff79906f5254).

- [x] First, we have to solve #1693 
- [x] [See updated Figma docs](https://www.figma.com/design/YDktJcseQIIQbsuCpoKS4V/Component-Docs?node-id=2265-6175&t=tZ9yNamTo7QVlkrU-11) which contain information of how to build toasts in an accessible way.

**We need to cover both requirements/notes in Figma and this ticket. In case we have conflicting suggestions/infos, we will prefer Figma.**

-----------

## Comment
**Problem:** Announcements are inconsistent, insufficient, and confusing.
- VoiceOver on macOs announces "Close"
- Talkback on Android announces "lorem ispum close".
- NVDA doesn't announce anything except when another message is already present; then it announces "notification, lorem ipsum close button" for the new message.
- JAWS announces, "notification, lorem ipsum close"

Screen reader users either don't receive the message or get a message that includes the button label, which some users could interpret as an action they must take.

One of the reasons for the inconsistency is that with some screen readers, live regions added to the page after the page has loaded are sometimes ignored. Another is that live regions don't announce roles, only text.

**Solution 1:** Use an empty wrapper div with `role=status` that's always present for your messages. Notifications that appear inside the div will be read out. Avoid interactive elements inside your notifications.
**Solution 2:** Don't use toasts.

**Problem:** People who use software to zoom into parts on the screen may miss toast messages.
**Solution 1:** Don't use toasts for critical information. Always show critical notifications inline on the page.
**Solution 2:** Don't use toasts.

**Problem:** When the page is zoomed in, toasts may obscure large parts of the screen.
**Solution 1:** Don't use toasts and show notifications only inline.

**Problem:** Keyboard users must tab through the remaining page to close toasts or use other controls within.
**Solution 1:** Don't move focus, don't use toasts for critical information and actions, and try to avoid using interactive elements in toasts.
**Solution 2:** Don't use toasts.

## Notes
- If someone were to ignore, or miss a toast message due to its timed display, there should be no negative impact on their current activities or the status that the message conveyed. Using the previous examples, ignoring a toast message would still mean that a file was saved, that a message was sent, or that a meeting was about to start. (Scott O’Hara)
- Consider not using buttons in timed toasts. [Scott O’Hara explains why](https://www.scottohara.me/blog/2019/07/08/a-toast-to-a11y-toasts.html#inclusive-ux-of-a-toast).

## Best Practices
- Notification shouldn’t close earlier than 5 seconds.
- Don’t use toasts for critical information.
- Prefer default (inline) notifications over toast notifications.

## Related WCAG criteria
- https://www.w3.org/WAI/WCAG22/quickref/#timing-adjustable
- https://www.w3.org/WAI/WCAG22/quickref/#reflow
- https://www.w3.org/WAI/WCAG22/quickref/#resize-text
- https://www.w3.org/WAI/WCAG22/quickref/#status-messages

## Additional resources
- https://www.scottohara.me/blog/2019/07/08/a-toast-to-a11y-toasts.html
- https://adrianroselli.com/2020/01/defining-toast-messages.html
- https://adamsilver.io/blog/the-problem-with-toast-messages-and-what-to-do-instead/

## DoR
- [ ] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

**Comment by karlbaumhauer** [comment]
@yoezlem: Design-research ticket to investigate UX on:
1. how is this component currently used and wich content is placed inside?
2. could we cover this need with pure inline-notification?

**Comment by yoezlem** [comment]
blocked by #1693
